### PR TITLE
Clarify sdfref based on discussion at 2020-12-14 interim.

### LINKED ISSUE #7: Issue: does sdfRef add or also override?
https://www.ietf.org/archive/id/draft-ietf-asdf-sdf-00.html#name-use-of-the-sdfref-keyword-t

asks:

ISSUE: Do we want to enable qualities from the source definition to be overridden in future versions?
The above only says "added".
(Yes, we do want to enable overriding, but need to warn specifiers not
to use this in a way that contradicts the referenced semantics.)

### PR DESCRIPTION
**Pull request by cabo**
Close #7.

### CODE REVIEW

File: sdf.md [hunk 1]
```diff
@@ -946,12 +946,25 @@ The structure of a path in a namespace is defined by the JSON Pointers to the de
...
-An existing definition may be used as a template for a new definition, that is, a new definition is created in the target namespace which uses the defined qualities of some existing definition. This pattern will use the keyword "sdfRef" as a quality of a new definition with a value consisting of a reference to the existing definition that is to be used as a template. Optionally, new qualities may be added and values of optional qualities and quality values may be defined.
 
-ISSUE: Do we want to enable qualities from the source definition to be overridden in future versions?
-The above only says "added".
-(Yes, we do want to enable overriding, but need to warn specifiers not
-to use this in a way that contradicts the referenced semantics.)
+An existing definition may be used as a template for a new definition, that is, a new definition is created in the target namespace which uses the defined qualities of some existing definition. This pattern will use the keyword "sdfRef" as a quality of a new definition with a value consisting of a reference to the existing definition that is to be used as a template. 
+
+In the same definition that uses "sdfRef", new qualities may be added
```
Comment (akeranen) [inline sdf.md:952]:
```suggestion
In the definition that uses "sdfRef", new qualities may be added
```

File: sdf.md [hunk 2]
```diff
@@ -946,12 +946,25 @@ The structure of a path in a namespace is defined by the JSON Pointers to the de
...
+and existing qualities from the referenced definition may be
+overridden.  (Note that JSON maps (objects) do not have a defined
+order, so the processor may see these overrides before seeing the
```
Comment (akeranen) [inline sdf.md:955]:
```suggestion
order, so the JSON processor may see these overrides before seeing the
```

File: sdf.md [hunk 3]
```diff
@@ -946,12 +946,25 @@ The structure of a path in a namespace is defined by the JSON Pointers to the de
...
+`sdfRef`.)
+
+As a convention, overrides are intended to be used only for further restricting
+the set of data values, as shown in {{exa-sdfref}}:  any value for a
+`cable-length` also is a valid value for a `length`, with the
+additional restriction that the length cannot be smaller than 5 cm.
+(This is labeled as a convention as it cannot be checked in the
+general case; a quality of implementation consideration for a tool
```
Comment (akeranen) [inline sdf.md:963]: I'd consider rewording `quality of implementation consideration for a tool`

**Review by akeranen — APPROVED** [review]

File: sdf.md (hunk 3 above)
Comment (cabo) [reply sdf.md:963]: Actually, this is carefully worded based on extensive discussions on previous documents.
What is it that you don't like?

Comment (akeranen) [reply sdf.md:963]: I found it hard to parse

**Review by mjkoster — APPROVED** [review]

File: sdf.md (hunk 2 above)
Comment (cabo) [reply sdf.md:955]: Actually, this is talking about the SDF processor, which may get the map entries handed in any order from the JSON processor.

File: sdf.md (hunk 3 above)
Comment (cabo) [reply sdf.md:963]: Well, it is a complicated concept.  Other specs often have made these considerations into interoperability requirements or did not mention them at all.  I'd rather say exactly what is meant here.

Comment (cabo) [reply sdf.md:963]: Well, OK, we use the word quality in a specific meaning here as well.  Ouch.
